src/potions.ts

```diff
@@ -434,3 +430,121 @@ export function bathroomFinance(embezzlers: number): void {
+  const potions = [
+    new VariableMeatPotions($item`love song of sugary cuteness`, 20, 2),
+    new VariableMeatPotions($item`pulled yellow taffy`, 50, 2),
+    // new VariableMeatPotions($item`porcelain candy dish`, 500, 1),
+  ];
+
+  potions.forEach((potion) => {
+    const barfTurns = Math.max(0, estimatedTurns() - yachtzees - embezzlers);
+    const yachtzeeValue = 2000;
+    const embezzlerValue = baseMeat + 750;
+    const barfValue = (baseMeat * turnsToNC) / 30;
+
+    function valueNPotions(n: number): number {
+      const totalCosts = n * potion.price(false);
+      const totalDuration = n * potion.duration;
+      let cappedDuration = Math.max(0, totalDuration - potion.softcap + 1);
+      let decayDuration = Math.min(totalDuration, potion.softcap - 1);
+      let totalValue = 0;
+      [
+        [yachtzees, yachtzeeValue],
+        [embezzlers, embezzlerValue],
+        [barfTurns, barfValue],
+      ].forEach((arr) => {
+        let turns = arr[0];
+        const value = arr[1];
+        if (cappedDuration >= turns) {
+          totalValue += (value * turns * potion.cappedMeatBonus) / 100;
+          cappedDuration -= turns;
+        } else {
+          totalValue += (value * cappedDuration * potion.cappedMeatBonus) / 100;
+          turns -= cappedDuration;
+          cappedDuration = 0;
+          if (decayDuration >= turns) {
+            totalValue +=
+              (value *
+                turns *
+                potion.meatBonusPerTurn *
+                triangleNumber(decayDuration, decayDuration - turns)) /
+              100;
+            decayDuration -= turns;
+          } else {
+            totalValue +=
+              (value * decayDuration * potion.meatBonusPerTurn * triangleNumber(decayDuration)) /
+              100;
+            decayDuration = 0;
+          }
+        }
+      });
+      return totalValue - totalCosts;
+    }
+
+    const potionAmountsToConsider = [
+      Math.floor((yachtzees + potion.softcap) / potion.duration),
+      Math.ceil((yachtzees + potion.softcap) / potion.duration),
+      Math.floor(yachtzees / potion.duration),
+      Math.ceil(yachtzees / potion.duration),
+      Math.floor((yachtzees + embezzlers + potion.softcap) / potion.duration),
+      Math.ceil((yachtzees + embezzlers + potion.softcap) / potion.duration),
+      Math.floor((yachtzees + embezzlers) / potion.duration),
+      Math.ceil((yachtzees + embezzlers) / potion.duration),
+      Math.floor((yachtzees + embezzlers + barfTurns + potion.softcap) / potion.duration),
+      Math.ceil((yachtzees + embezzlers + barfTurns + potion.softcap) / potion.duration),
+      Math.floor((yachtzees + embezzlers + barfTurns) / potion.duration),
+      Math.ceil((yachtzees + embezzlers + barfTurns) / potion.duration),
+    ];
+    const profits = potionAmountsToConsider.map(valueNPotions);
+    const bestProfits = profits.reduce((a, b) => (a < b ? a : b));
+    const bestIdx = profits.indexOf(bestProfits);
```

Comment:
I don't think this is necessary; the way bestProfits is constructed, we _can't_ get fellas from outside the array into it